compiler/rustc_resolve/src/imports.rs

```diff
@@ -323,7 +324,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                             // imported from the same glob-import statement.
                             resolution.binding = Some(binding);
                         } else if res != old_binding.res() {
-                            let binding = if warn_ambiguity {
+                            let binding = if old_binding.is_ambiguity_in_extern_crate() {
```

Comment:
What if the *new* binding is an ambiguity in extern crate?
In that case we should probably report a warning instead of an error too.